posthog/api/capture.py

```diff
@@ -68,7 +68,16 @@ def log_event(data: Dict, event_name: str, partition_key: Optional[str]):
 
     # TODO: Handle Kafka being unavailable with exponential backoff retries
     try:
-        future = KafkaProducer().produce(topic=KAFKA_EVENTS_PLUGIN_INGESTION_TOPIC, data=data, key=partition_key)
+        future = KafkaProducer().produce(
+            topic=KAFKA_EVENTS_PLUGIN_INGESTION_TOPIC,
+            data=data,
+            key=partition_key,
+            headers={
+                "team_id": data.get("team_id"),
+                "api_token": data.get("token"),
```

Comment:
do we validate these aren't 1M chars or garbage?

Reply:
No, although these tokens are that we generate I gather? So if it is 1M chars and over the header size limit then I guess we would want to drop it anyway?

Reply:
Worth checking the shape of the token to reject ones that are obviously not a valid one.
We probably have several formats, from older versions?

Reply:
I believe the tests are already covering the available formats, and I'm a little wary about going down the road of adding validation to the capture rest interface in this PR. I don't think I'm introducing any new assumptions about `token` here.

Separately I would love to 1. add validation on the API interface and 2. properly define the interface in e.g. OpenAPI spec.

When I merge this and we start 500 erroring I'll eat my words 😆

Reply:
In its current state, we'll only write to kafka if either:
  - the token is valid in pg
  - it's in the LIGHTWEIGHT_CAPTURE_ENDPOINT_ENABLED_TOKENS allowlist

So it's safe to ignore this now That's something we will have to do before we completely roll-out pg-less capture later on.